src/BoundingBoxCameraSensor.cc

```diff
@@ -0,0 +1,532 @@
+  this->dataPtr->boundingboxCamera->SetFarClipPlane(sdfCamera->FarClip());
+  this->dataPtr->boundingboxCamera->SetImageFormat(PixelFormat::PF_R8G8B8);
+  this->dataPtr->boundingboxCamera->SetAspectRatio(aspectRatio);
+  this->dataPtr->boundingboxCamera->SetHFOV(angle);
+  this->dataPtr->boundingboxCamera->SetVisibilityMask(
+    sdfCamera->VisibilityMask());
+  this->dataPtr->boundingboxCamera->SetBoundingBoxType(this->dataPtr->type);
+
+  // Add the camera to the scene
+  this->Scene()->RootVisual()->AddChild(this->dataPtr->camera);
+  this->Scene()->RootVisual()->AddChild(this->dataPtr->boundingboxCamera);
+
+  // Add the rendering sensor to handle its render, no need to add the
+  // rgb camera as we handle its render via Capture() function
+  this->AddSensor(this->dataPtr->boundingboxCamera);
+
+  // Connection to receive the BoundingBox buffer
+  this->dataPtr->newBoundingBoxConnection =
+    this->dataPtr->boundingboxCamera->ConnectNewBoundingBoxes(
+      std::bind(&BoundingBoxCameraSensor::OnNewBoundingBoxes, this,
+        std::placeholders::_1));
+
+  this->dataPtr->image = this->dataPtr->camera->CreateImage();
+
+  return true;
+}
+
+/////////////////////////////////////////////////
+rendering::BoundingBoxCameraPtr BoundingBoxCameraSensor::BoundingBoxCamera()
+{
+  return this->dataPtr->boundingboxCamera;
+}
+
+/////////////////////////////////////////////////
+void BoundingBoxCameraSensor::OnNewBoundingBoxes(
+  const std::vector<rendering::BoundingBox> &boxes)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+  this->dataPtr->boundingBoxes.clear();
+  for (auto box : boxes)
+    this->dataPtr->boundingBoxes.push_back(box);
+}
+
+//////////////////////////////////////////////////
+bool BoundingBoxCameraSensor::Update(
+  const std::chrono::steady_clock::duration &_now)
+{
+  IGN_PROFILE("BoundingBoxCameraSensor::Update");
+  if (!this->dataPtr->initialized)
+  {
+    ignerr << "Not initialized, update ignored.\n";
+    return false;
+  }
+
+  if (!this->dataPtr->boundingboxCamera || !this->dataPtr->camera)
+  {
+    ignerr << "Camera doesn't exist.\n";
+    return false;
+  }
+
+  // don't render if there is no subscribers
+  if (!this->dataPtr->imagePublisher.HasConnections() &&
+    !this->dataPtr->boxesPublisher.HasConnections())
+  {
+    return false;
+  }
+
+  // Render the bounding box camera
+  this->Render();
+
+  // Publish image only if there is subscribers for it
+  this->dataPtr->IsPublishedImage =
+    this->dataPtr->imagePublisher.HasConnections();
+
+  if (this->dataPtr->IsPublishedImage)
+  {
+    // The sensor updates only the bounding box camera with its pose
+    // as it has the same name, so make rgb camera with the same pose
+    this->dataPtr->camera->SetWorldPose(
+      this->dataPtr->boundingboxCamera->WorldPose());
+
+    // Render the rgb camera
+    this->dataPtr->camera->Capture(this->dataPtr->image);
+
+    // Draw bounding boxes
+    for (auto box : this->dataPtr->boundingBoxes)
```

Comment:
```suggestion
    for (const auto &box : this->dataPtr->boundingBoxes)
```

Reply:
cannot do that here, since it is refused by the function DrawBoundingBox

Reply:
Why would it not work with `DrawBoundingBox`? That method takes a const reference:
https://github.com/ignitionrobotics/ign-rendering/blob/653d60af0a1f38ddf67b55392350cb739fbad2b1/include/ignition/rendering/BoundingBoxCamera.hh#L104